workers/unity/Packages/com.improbable.gdk.gameobjectcreation/GameObjectInitializationSystem.cs

```diff
@@ -0,0 +1,109 @@
+using System.Collections.Generic;
+using Unity.Collections;
+using Unity.Entities;
+using UnityEngine;
+
+namespace Improbable.Gdk.Core.GameObjectRepresentation
+{
+    /// <summary>
+    ///     Creates a companion gameobject for newly spawned entities according to a prefab definition.
+    /// </summary>
+    [UpdateInGroup(typeof(SpatialOSReceiveGroup.EntityInitialisationGroup))]
+    [DisableAutoCreation]
+    internal class GameObjectInitializationSystem : ComponentSystem
+    {
+        private struct AddedEntitiesData
+        {
+            public readonly int Length;
+            public EntityArray Entities;
+            [ReadOnly] public ComponentDataArray<SpatialEntityId> SpatialEntityIds;
+            [ReadOnly] public ComponentDataArray<NewlyAddedSpatialOSEntity> NewlyCreatedEntities;
+        }
+
+        private struct RemovedEntitiesData
+        {
+            public readonly int Length;
+            public EntityArray Entities;
+            [ReadOnly] public ComponentDataArray<GameObjectReferenceHandle> GameObjectReferenceHandles;
+            public SubtractiveComponent<GameObjectReference> NoGameObjectReference;
+        }
+
+        [Inject] private AddedEntitiesData addedEntitiesData;
+        [Inject] private RemovedEntitiesData removedEntitiesData;
+
+        [Inject] private EntityGameObjectLinkerSystem LinkerSystem;
+
+        private WorkerSystem worker;
+        private ViewCommandBuffer viewCommandBuffer;
+        private readonly Dictionary<Entity, GameObject> entityGameObjectCache = new Dictionary<Entity, GameObject>();
+        private readonly IEntityGameObjectCreator gameObjectCreator;
+
+        public GameObjectInitializationSystem(IEntityGameObjectCreator gameObjectCreator)
+        {
+            this.gameObjectCreator = gameObjectCreator;
+        }
+
+        protected override void OnCreateManager(int capacity)
+        {
+            base.OnCreateManager(capacity);
+
+            worker = World.GetExistingManager<WorkerSystem>();
```

Comment:
Should this be `[Inject]`ed?

Reply:
can place it in one of the TODO tickets

Reply:
yeah, should be. But can also be done in my PR. whatever Peter prefers